dss_metrics/minio_rest_collector.py

```diff
@@ -64,24 +64,28 @@ def poll_statistics(self, metrics_data_buffer, exit_flag):
                 if not endpts or len(endpts) == 0:
                     continue
 
-                minio_endpoint = endpts.pop()
-                miniocluster_id = self.get_minio_cluster_uuid(minio_endpoint)
-                minio_metrics = self.get_minio_metrics_from_endpoint(
-                    minio_endpoint)
-
-                tags = {}
-                tags['cluster_id'] = self.cluster_id
-                tags['target_id'] = socket.gethostname()
-                tags['minio_id'] = miniocluster_id
-                tags['type'] = self.TYPE
-
-                for metric in minio_metrics:
-                    if self.filter and not self.check_whitelist_key(metric[0]):
-                        continue
-                    metrics_data_buffer.append(
-                        metrics.MetricInfo(
-                            metric[0], metric[0], metric[1], tags, time.time())
-                    )
+                for minio_endpoint in endpts:
+                    miniocluster_id = self.get_minio_cluster_uuid(
+                        minio_endpoint)
+                    minio_metrics = self.get_minio_metrics_from_endpoint(
+                        minio_endpoint)
+
+                    tags = {}
+                    tags['cluster_id'] = self.cluster_id
+                    tags['target_id'] = socket.gethostname()
+                    tags['minio_id'] = miniocluster_id
+                    tags['minio_endpoint'] = minio_endpoint
+                    tags['type'] = self.TYPE
+
+                    for metric in minio_metrics:
+                        if (self.filter and not
+                                self.check_whitelist_key(metric[0])):
+                            continue
+                        metrics_data_buffer.append(
+                            metrics.MetricInfo(
+                                metric[0], metric[0], metric[1], tags,
```

Comment:
Why pass metrics selectively?

Reply:
It is because we may not want to expose all metrics to the end user all of the time, hence why we filter using a whitelist if the filter flag is given

Reply:
I mean, why not `metrics.MetricInfo(metric, tags..)` ? Let `MetricInfo` filter out.
`metrics.MetricInfo(metric[0], metric[0], metric[1], tags,` doen't really help readability, else its better to use named params.

Reply:
See new commits on PR, I made a lot of changes and not using this approach anymore